### DISCUSSION TITLE
GPS recording

### DISCUSSION THREAD

**Discussion by airald** [devemux86/cruiser#419] [Q&A] [answered]
I use GPS recording usually on my motorcycle profile. I use on route and following mode. It's very useful because it stops automatically when I do a stop and turn off Cruiser. 

Recently I'm using also for walking. In this situation, most of the time I don't have a route and don't use follow. So I start recording manually with "GPS" on my right panel.

I don't know why sometimes I put Cruiser on foreground and recording is stopped (Cruiser is running on background, there's the icon on notif bar). Sometimes I think that's because I've been not moving some time, but I'm not sure. 

Can anybody describe the behaviour of this feature? I've been searching the forum with no success.

**Comment by devemux86** [comment]
@airald thanks for the report.

To clarify, with motorcycle you use the "Settings | GPS recording | Automatic recording of track"
and have selected "Navigation" and "Follow location"?
And these modes work properly, even with Cruiser in background?

Now the question is when we walk and manually start GPS recording,
outside of these automatic modes?

> recording is stopped

What do you mean "stopped"?
What icon does the GPS recording panel has? The record (circle) or the stop icon?

Has it stopped or it just cannot record because of the active filters?
What filters do you use? Maybe walking needs different filter options?

**Reply by airald** [reply]
When recording on Navigation and Follow location Cruiser records properly even in background.

When I start recording manually, sometimes I put Cruiser on foreground and the icon is the red circle, so I assume that recording is stopped.

The filters I use are 5 seconds between points, 5 meters minimum distance, 50 meters min accuracy and speed threshold 1 km/h.

**Reply by devemux86** [reply]
> When I start recording manually, sometimes I put Cruiser on foreground and the icon is the red circle, so I assume that recording is stopped.

After you started manually the GPS recording, have you started + **stopped** follow location or navigation?

If "Automatic recording of track" is enabled and stop follow or navigation,
then GPS recording stops automatically.

**Reply by airald** [reply]
I know that start/stop follow location stops recording. I'm not conscious to touch anything on the screen. I will do some more tests, I just want to know If I'm doing something wrong. 

I'll do more tests to try to find the reason or the conditions.

Thank's

**Reply by airald** [reply]
Yesterday I started recording while I was at the office. No movement, I put on foreground many times and the icon was PAUSE, so cruiser was recording. I used the phone several times. When I go to lunch 3 km far, I put on foreground after 15 minutes aprox stop my travel. I see the trace on the screen, but the icon was the two circles, recording was stopped. I started again. I return to the office and I repeat the same operation. Cruiser recorded the return and stop recording. I started again. The same when going home.

This morning I go out and walk about 30 meters with cruiser in background and return to the office. When I just enter at the office, I see that Cruiser recorded all my "travel" and recording stopped.

It happened several times on different situations. Maybe could be when there's no GPS signal... is present on all tests. In some tests I arrived at home or office where phone has wi-fi, but not on other tests, like yesterday for lunch time, I move not in my car to a non wi-fi place, so it's equivalent to walk.

It happens always that I arrive a non GPS signal location.

**Reply by devemux86** [reply]
@airald thanks for the detailed tests and explanation! :+1: 

Let me check the code and see if there are any cases where the GPS recording might (wrongly) stop.

**Comment by devemux86** [comment]
@airald when you saw recording stop (circle icon), was the notification (arrow icon) at the top?

What device and Android do you use?
Have you excluded Cruiser from battery optimization in Android settings?

Please see here: https://dontkillmyapp.com

**Reply by devemux86** [reply]
@airald also when you put Cruiser again in the foreground, did it really come to the foreground (appear immediately)?

Or it started again (slower start)? (then it was closed by Android)

**Reply by devemux86** [reply]
@airald also if you do not put Cruiser in background and make similar tests with the app in foreground,
does the GPS recording stops?

**Reply by airald** [reply]
Cruiser icon is always on notification. I have MIUI 14 and Cruiser is running with any energy optimisation. I can keep Cruiser in background hours with no closes. When I put on foreground was really on background, it appears faster. Cruiser icon is on notification bar and also appears a "1" on my desktop icon when recording is enabled.

I will repeat the test later.

**Reply by devemux86** [reply]
> Cruiser is running with any energy optimisation. I can keep Cruiser in background hours with no closes.

This is good news! :+1: 

> also appears a "1" on my desktop icon when recording is enabled

You probably mean the notification dot (may be different on Xiaomi):

![Cruiser](https://github.com/devemux86/cruiser/assets/3484020/7913cd15-2650-49b7-bdf4-1b22758b9894)

**Reply by airald** [reply]
Yes, I use NOVA as launcher, but is similar, with the number of notifications inside the dot.

![image](https://github.com/devemux86/cruiser/assets/20151719/1eff7d73-8d8f-484c-b979-fd91e530ec07)

**Reply by devemux86** [reply]
@airald can you send me the profile of your tests?

So I can check the settings and try to reproduce it.

**Reply by airald** [reply]
Cruiser on foreground but screen locked causes also record stop

Cruiser on foreground with screen on keeps recording on. (tested one time)
After see that, I put Cruiser on background (while I was with no GPS signal), automatically stopped recording.

I will test it with another device (a Samsung Tablet).

The strange thing is that not happens when I'm recording normally.

Attached the profile used.

[Senderisme.json](https://github.com/devemux86/cruiser/files/13888078/Senderisme.json)

**Reply by devemux86** [reply]
@airald thanks for the profile.

> The strange thing is that not happens when I'm recording normally.

Do you mean the automatic recording in navigation or follow location?

**Reply by airald** [reply]
I mean when Cruiser is recording points because position is changing. When I use navigation, always have the screen on with cruiser on foreground.

**Reply by devemux86** [reply]
@airald can you try with a _clean_ profile?

So we can see if an enabled setting is causing this behavior.
(make a copy of your profile and reset it)

**Comment by devemux86** [comment]
Cruiser only has access to your location "**while using the app**" (background or foreground).
(see location permission, similar to other navigators)

It does not access your location "**all the time**" like other apps, this is important for **privacy**!

<img src="https://github.com/devemux86/cruiser/assets/3484020/590b94f0-e723-4eb7-acf0-175ffdaa02a7" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/a5127f98-4d8c-418e-a967-4d3c92e88ab3" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/60598a42-d6bf-4632-aa5e-54a1370368bf" width="33%"/>

**Reply by airald** [reply]
location is selected "allow only while using the app"

**Comment by devemux86** [comment]
@airald I'm trying to reproduce the steps, are they correct?

- Turn off GPS
- Start Cruiser > Start GPS recording > Put Cruiser in background
- Lock phone / Close screen

Open phone: Cruiser's notification is on top bar and notification dot on desktop.
Resume Cruiser > GPS recording is on.

---

Can anyone else test with their devices?

**Reply by Ciutnik** [reply]
I followed the above steps on Samsung. GPS recording reported as active.

**Reply by airald** [reply]
I tested also on my Samsung tab and the result is the same as @Ciutnik , so it seems to be a MIUI issue.

If I turn OFF GPS on my Redmi, there's no problem, recording is active. 

I tested also on my old Redmi 10, recording is NOT stopped. Both Redmi 10 and Redmi 11 are running MIUI 14 and the same Cruiser profile.

**Reply by devemux86** [reply]
@airald do they have the same battery optimization settings?

Can you make some tests with the battery optimization settings to see if they close the app or the GPS?

The operating system might somehow close the app? I cannot find in the code any stop of the recording.
(except the automatic stop for navigation or follow location - _when we have it enabled_)

**Comment by airald** [comment, marked as answer]
After a lot of tests, I finally found where is the problem.

I use Automate to change phone settings when I arrive or leave places (home, office, car...) One of the things that do is change wallpaper. If I change wallpaper (manually or through Automate) when screen is off, then makes cruiser "close". This particular conditions make more difficult to find the problem.

It seems that Cruiser is running because notifications are still here. If I close all apps, notifications also closes. Maybe Cruiser is not closed but closes and re-opens when I try to put on foreground.

Very strange, but I tested a lot of times in my two Redmi and all tests follows this behaviour. I tested the same on Samsung and nothing happened to cruiser, continues recording.

I started Cruiser yesterday night recording and I stopped it this afternoon. All my travel has been recordered perfectly, moving on car and also walking.

Thank's @devemux86 and @Ciutnik to help to find the problem.

**Reply by devemux86** [reply]
@airald many thanks for the detailed tests and the explanation of what happens! :+1: 

About the notification, maybe the operating system keeps and resends it during quick app restart.

Cruiser is highly optimized for performance and low battery, so restarts may feel like a resume. :slightly_smiling_face:
